File: nan21.dnet.module.bd/nan21.dnet.module.bd.ui.extjs/src/main/webapp/src/ds/LocationDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.bd.geo.ds.model.LocationDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"entityFQN", type:"string"},
		{name:"regionName", type:"string"},
		{name:"cityName", type:"string"},
		{name:"adress", type:"string"},
		{name:"notes", type:"string"},
		{name:"active", type:"boolean"},
		{name:"countryId", type:"int", useNull:true},
		{name:"countryCode", type:"string"},
		{name:"regionId", type:"int", useNull:true},
		{name:"regionCode", type:"string"},
		{name:"cityId", type:"int", useNull:true},
		{name:"shipping", type:"boolean"},
		{name:"billing", type:"boolean"},
		{name:"mailing", type:"boolean"},
		{name:"targetUuid", type:"string"},
		{name:"targetType", type:"string"}
	]
});

Ext.define("net.nan21.dnet.module.bd.geo.ds.model.LocationDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"regionName", type:"string"},
		{name:"cityName", type:"string"},
		{name:"adress", type:"string"},
		{name:"notes", type:"string"},
		{name:"active", type:"boolean", useNull:true},
		{name:"countryId", type:"int", useNull:true},
		{name:"countryId_From",type:"int", useNull:true},
		{name:"countryId_To",type:"int", useNull:true},
		{name:"countryCode", type:"string"},
		{name:"regionId", type:"int", useNull:true},
		{name:"regionId_From",type:"int", useNull:true},
		{name:"regionId_To",type:"int", useNull:true},
		{name:"regionCode", type:"string"},
		{name:"cityId", type:"int", useNull:true},
		{name:"cityId_From",type:"int", useNull:true},
		{name:"cityId_To",type:"int", useNull:true},
		{name:"shipping", type:"boolean", useNull:true},
		{name:"billing", type:"boolean", useNull:true},
		{name:"mailing", type:"boolean", useNull:true},
		{name:"targetUuid", type:"string"},
		{name:"targetType", type:"string"}
	] 
});
	
Ext.define("net.nan21.dnet.module.bd.geo.ds.param.LocationDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 

File: nan21.dnet.module.bd/nan21.dnet.module.bd.presenter/src/main/java/net/nan21/dnet/module/bd/geo/ds/model/LocationDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.bd.geo.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;
import net.nan21.dnet.module.bd.geo.domain.entity.Location;

@Ds(entity = Location.class)
public class LocationDs extends AbstractAuditableDs<Location> {

    public static final String fREGIONNAME = "regionName";
    public static final String fCITYNAME = "cityName";
    public static final String fADRESS = "adress";
    public static final String fNOTES = "notes";
    public static final String fACTIVE = "active";
    public static final String fCOUNTRYID = "countryId";
    public static final String fCOUNTRYCODE = "countryCode";
    public static final String fREGIONID = "regionId";
    public static final String fREGIONCODE = "regionCode";
    public static final String fCITYID = "cityId";
    public static final String fSHIPPING = "shipping";
    public static final String fBILLING = "billing";
    public static final String fMAILING = "mailing";
    public static final String fTARGETUUID = "targetUuid";
    public static final String fTARGETTYPE = "targetType";

    @DsField()
    private String regionName;

    @DsField()
    private String cityName;

    @DsField()
    private String adress;

    @DsField()
    private String notes;

    @DsField()
    private Boolean active;

    @DsField(join = "left", path = "country.id")
    private Long countryId;

    @DsField(join = "left", path = "country.code")
    private String countryCode;

    @DsField(join = "left", path = "region.id")
    private Long regionId;

    @DsField(join = "left", path = "region.code")
    private String regionCode;

    @DsField(join = "left", path = "city.id")
    private Long cityId;

    @DsField(path = "forShipping")
    private Boolean shipping;

    @DsField(path = "forBilling")
    private Boolean billing;

    @DsField(path = "forMailing")
    private Boolean mailing;

    @DsField()
    private String targetUuid;

    @DsField()
    private String targetType;

    public LocationDs() {
        super();
    }

    public LocationDs(Location e) {
        super(e);
    }

    public String getRegionName() {
        return this.regionName;
    }

    public void setRegionName(String regionName) {
        this.regionName = regionName;
    }

    public String getCityName() {
        return this.cityName;
    }

    public void setCityName(String cityName) {
        this.cityName = cityName;
    }

    public String getAdress() {
        return this.adress;
    }

    public void setAdress(String adress) {
        this.adress = adress;
    }

    public String getNotes() {
        return this.notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public Boolean getActive() {
        return this.active;
    }

    public void setActive(Boolean active) {
        this.active = active;
    }

    public Long getCountryId() {
        return this.countryId;
    }

    public void setCountryId(Long countryId) {
        this.countryId = countryId;
    }

    public String getCountryCode() {
        return this.countryCode;
    }

    public void setCountryCode(String countryCode) {
        this.countryCode = countryCode;
    }

    public Long getRegionId() {
        return this.regionId;
    }

    public void setRegionId(Long regionId) {
        this.regionId = regionId;
    }

    public String getRegionCode() {
        return this.regionCode;
    }

    public void setRegionCode(String regionCode) {
        this.regionCode = regionCode;
    }

    public Long getCityId() {
        return this.cityId;
    }

    public void setCityId(Long cityId) {
        this.cityId = cityId;
    }

    public Boolean getShipping() {
        return this.shipping;
    }

    public void setShipping(Boolean shipping) {
        this.shipping = shipping;
    }

    public Boolean getBilling() {
        return this.billing;
    }

    public void setBilling(Boolean billing) {
        this.billing = billing;
    }

    public Boolean getMailing() {
        return this.mailing;
    }

    public void setMailing(Boolean mailing) {
        this.mailing = mailing;
    }

    public String getTargetUuid() {
        return this.targetUuid;
    }

    public void setTargetUuid(String targetUuid) {
        this.targetUuid = targetUuid;
    }

    public String getTargetType() {
        return this.targetType;
    }

    public void setTargetType(String targetType) {
        this.targetType = targetType;
    }

}
